src/core/thread/mle.cpp

```diff
@@ -246,6 +252,7 @@ ThreadError Mle::Stop(void)
         mNetif.RemoveUnicastAddress(mLeaderAloc);
     }
 
+    mNetif.GetPendingDataset().StopDelayTimer();
```

Comment:
The starting Delay Timer action in `SetStateChild()` and `SetStateRouter()` is to address the following case: the device detached due to `thread stop` command, while the network datasets are still exist, and then after calling `thread start`, the device will re-sync/reattach to network without refreshing Pending Dataset since it already have the same network datasets, this way the timer is not started, so I check the Delay Timer state here and restart it if it is not running.
I have also consider to clear network datasets (`mNetwork`) in `Mle::Stop()`, to ensure the device could refresh its network datasets during reattach process, so then we don't need to check and restart Delay Timer in `SetStateChild()` and `SetStateRouter()`, do you think this way is better?